docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,104 @@
+1	2
+```
+```
+$ ls /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/1
+1.root.json	snapshot.json	targets.json	timestamp.json
+```
+
+And now we can also see that after bumping the version and moving timestamp v2 expiry date two weeks forward from v1, the v2 corresponding timestamp metadata has recorded that expiry date correctly:
+
+Timestamp v1:
+<pre><code>$ cat /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/1/timestamp.json 
+{
+ "signatures": [{...}],
+ "signed": {
+  "_type": "timestamp",
+  <b>"expires": "2022-03-30T00:18:31Z"</b>,
+  "meta": { "snapshot.json": {"version": 1}},
+  "spec_version": "1.0.28",
+  "version": 1
+ }}
+</code></pre>
+
+Timestamp v2:
+
+<pre><code>$ cat /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/2/timestamp.json 
+{
+ "signatures": [{...}],
+ "signed": {
+  "_type": "timestamp",
+  <b>"expires": "2022-04-13T00:18:31Z"</b>,
+  "meta": { "snapshot.json": {"version": 2}},
+  "spec_version": "1.0.28",
+  "version": 2
+ }}
+</code></pre>
+
+## Edge cases are a norm when we talk about security
+
+The TUF spec says the following:
+“Record the time at which the update began as the fixed update start time. Time is fixed at the beginning of the update workflow to allow an application using TUF to effectively pause time, in order to ensure that metadata which has a valid expiration time at the beginning of an update does not fail an expiration check later in the update workflow.”
+
+Now imagine that we have performed an update and stored metadata in a cache. And the locally stored timestamp/snapshot has expired. But we still need it to perform an update from remote by verifying the signatures and we need to use the expired timestamp. Here the repository simulator can do a great job testing this scenario, as we’d like to somehow update expiry without explicitly touching the file system from the tests.
+
+### Example 1:
+We can play with versions and expiry to verify that this scenario not explicitly mentioned in the spec works correctly and safely. By using the simulator, we can do the following:
+* Set the timestamp expiry one week ahead (Day 7)
+* On the very first day download, verify, and load metadata for the [top-level roles](https://theupdateframework.github.io/specification/latest/#roles-and-pki) following the specified order and implementing all the checks required in the TUF client workflow (which we simply call a refresh)
+* Then we bump [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) versions to v2 on the same day and set v2 expiry dates three weeks ahead (Day 21)
+
+Then if we travel in time somewhere between Day 7 and Day 21, we’d like to perform a successful refresh with the expired locally cached timestamp and we want the final version of the snapshot and targets roles to be v2.
```

Comment:
I think you can make the last sentence bullets as well and use numbers instead of just bullets.
Then you can add comments referencing the steps.
Also, I think you can split some of the bulltes.
For example, you can do something like this:
```suggestion
1. Set the timestamp expiry one week ahead (day 7).
2. On day 1 download, verify, and load metadata for the [top-level roles](https://theupdateframework.github.io/specification/latest/#roles-and-pki) following the TUF specification order. This is done by simply calling `RespositorySimulator.refresh()`.
3.  On the same day, set [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) expiration date three weeks ahead (day 21).
4. Bump [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) versions to v2 on the same day.
5. Travel in time somewhere between day 7 and day 21.
6. Perform a successful `refresh` (with `RepositorySimulator.refresh()` call).
7. Check that the final version of [snapshot].(https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) is 2.

Keep in mind that when performing the second `refresh` on step 6 [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) has expired, but still, we manage to make a successful refresh to version 2.
```

Reply:
Thank you! I like your proposal, looks easier to understand.